mantidproject/mantid
# Redesign peaks viewer to draw mixed shapes and ellipsoids

Fixes #11744 
Fixes #7987 

**Context**

This ticket aims to unify the peak representation in the SliceViewer. Currently the way peaks are represented is in large parts duplicated and different peak types cannot be present within a single peak workspace. In addition the user cannot plot elliptical peaks, which is made possible here. 

**To test:**

Please find sample data sets and installers (for OSX and Ubuntu) here: //olympic/babylon5/Scratch/Anton/SliceViewer/PR_15490

There are three test scenarios: for no-shape, spherical and elliptical. Follow the test instructions but please also perform free testing.

**Test peaks without any shape**

Required data sets: SXD_NaCl_NKL.nxs (signal) and MDPeaks_NaCl_1.nxs (peaks)
1. Load the two data sets into the Mantid
2. Open the SliceViewer with SXD_NaCl_NKL.nxs
3. Drag and drop the MDPeaks_NaCl_1.nxs onto the SliceViewer interface
   - Confirm that the peaks are visible in form of a cross
4. Select a peak from the peaks table
   - Confirm that the view is being zoomed to the selected peak
5. Press the color button for the the cross and select a different color
   - Confirm that the peak marker color changes

**Test spherical peaks**

Required data sets: SphericalPeaksData.nxs (signal) and SphericalPeaks.nxs (peaks)
1. Load the two data sets into the Mantid
2. Open the SliceViewer with SphericalPeaksData.nxs
3. Drag and drop the  SphericalPeaks.nxs onto the SliceViewer interface
   - Confirm that the peaks are visible in form of a circle
4. Select a peak from the peaks table
   - Confirm that the view is being zoomed to the selected peak
5. Press the color button for the the circle and select a different color
   - Confirm that the peak marker color changes
6. Press the background color button for the circle and select a different color. Select the "Show Background" option
   - Confirm that the peak marker background color changes

**Test elliptical peaks**

Required data sets: EllipticalPeaksData.nxs (signal) and EllipticalPeaks.nxs (peaks)
1. Load the two data sets into the Mantid
2. Open the SliceViewer with EllipticalPeaksData.nxs
3. Drag and drop the  EllipticalPeaks.nxs onto the SliceViewer interface
   - Confirm that the peaks are visible in form of an ellipse
4. Select a peak from the peaks table
   - Confirm that the view is being zoomed to the selected peak
5. Press the color button for the the ellipse and select a different color
   - Confirm that the peak marker color changes
6. Press the background color button for the ellipse and select a different color. Select the "Show Background" option
   - Confirm that the peak marker background color changes


### Release notes

See [here](http://www.mantidproject.org/index.php?title=ReleaseNotes_3_7_UI_Changes&diff=26380&oldid=26341)

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
